### PR TITLE
Add CryptoSigner.private_bytes

### LINKED ISSUE #798: CryptoSigner should expose private key bytes
Based on a couple of examples (the CRYPTOSIGNER.md doc and the example code in python-tuf) I think we should have a 
```
@property
def private_bytes(self) -> bytes
```
in CryptoSigner.

I'll make a PR

### PR DESCRIPTION
**Pull request by jku**
IMO this completes the MVP CryptoSigner API

Fixes #798

This is what the docs will look like: https://github.com/secure-systems-lab/securesystemslib/blob/cc68f4d1961da2bf274eab8b64baf4339ee06827/docs/CRYPTO_SIGNER.md

### CODE REVIEW

**Comment by jku** [comment]
rebased on main

File: tests/test_signer.py [hunk 1]
```diff
@@ -703,6 +703,23 @@ def test_generate(self):
             with self.assertRaises(UnverifiedSignatureError):
                 signer.public_key.verify_signature(sig, b"NOT DATA")
 
+    def test_private_bytes(self):
+        """Test full cycle of generate -> private_bytes -> from_priv_key_uri"""
+        generate_funcs = [
+            CryptoSigner.generate_rsa,
+            CryptoSigner.generate_ecdsa,
+            CryptoSigner.generate_ed25519,
+        ]
```
Comment (lukpueh) [inline tests/test_signer.py:712]: nit: Generate is already tested above. If we want to optimize test speed, we could use the existing test keys in PEMS_DIR / f"{keytype}_private.pem" here.

Comment (jku) [reply tests/test_signer.py:712]: yeah I guess some RSA keygen does take some time...

File: tests/test_signer.py [hunk 2]
```diff
@@ -703,6 +703,21 @@ def test_generate(self):
...
 
+    def test_private_bytes(self):
+        """Test private_bytes -> from_priv_key_uri"""
+        for pem in ["rsa", "ecdsa", "ed25519"]:
+            with open(PEMS_DIR / f"{pem}_private.pem", "rb") as f:
+                privkey = load_pem_private_key(f.read(), None)
+                signer = CryptoSigner(privkey)
+
+            with open("privkey.pem", "wb") as f:
+                f.write(signer.private_bytes)
```
Comment (lukpueh) [inline tests/test_signer.py:714]: Should we write this to a temp/test dir? Otherwise this litters cwd.

Comment (jku) [reply tests/test_signer.py:714]: oh true. I wish this was a default unittest thing...

**Review by lukpueh — APPROVED** [review]
Nice! I think you should fix the test, to not leave behind a pem. Otherwise this looks great

**Pull request merged**